File: recommendation-engine/hashing.py

def hash(arr):
  return "#".join(arr)

def unhash(hash):
  return hash.split("#")

def compare(list1, list2):
  intersection = len(list(set(list1).intersection(list2)))
  union = (len(set(list1)) + len(set(list2))) - intersection
  return float(intersection) / union

File: functions/src/hashing.ts
const jaccard = require('jaccard');

exports.hash = function(arr: string[]) {
  return arr.join("#").toUpperCase();
}

exports.unhash = function(hash: string) {
  return hash.split("#")
}

exports.compareHash = function(h1: string, h2: string) {
  const h1_unhashed = exports.unhash(h1);
  const h2_unhashed = exports.unhash(h2);
  return jaccard.index(h1_unhashed, h2_unhashed);
}

exports.compareSet = function(s1: string[], s2: string[]) {
  return jaccard.index(s1, s2);
}